packages/keyring-controller/src/KeyringController.ts

```diff
@@ -726,8 +711,8 @@ export class KeyringController extends BaseController<
 
   async restoreQRKeyring(serialized: any): Promise<void> {
     (await this.getOrAddQRKeyring()).deserialize(serialized);
+    await this.#keyring.persistAllKeyrings();
```

Comment:
If we don't persist keyrings after deserializing accounts on a Keyring directly, we don't add these eventual accounts to the vault. Also, we don't receive any update from `EthKeyringController`, thus we'll not sync keyrings and accounts when calling `restoreQRKeyring`

Reply:
Just curious — did this problem exist before or was it created by this PR? I assume the latter, but I'm unclear on how.

Reply:
This problem has always been there, but it has been noticeable just after my tests refactoring as we now assert on the internal state